core/templates/pages/exploration-editor-page/statistics-tab/statistics-tab.directive.ts

```diff
@@ -60,61 +58,79 @@ angular.module('oppia').directive('statisticsTab', [
         'DateTimeFormatService', 'ExplorationDataService',
         'ExplorationFeaturesService', 'ExplorationStatesService',
         'ReadOnlyExplorationBackendApiService', 'RouterService',
-        'StateImprovementSuggestionService',
-        'StateInteractionStatsService', 'StatesObjectFactory',
-        'UrlInterpolationService', 'IMPROVE_TYPE_INCOMPLETE',
+        'StateImprovementSuggestionService', 'StateInteractionStatsService',
+        'StatesObjectFactory', 'UrlInterpolationService',
+        'IMPROVE_TYPE_INCOMPLETE',
         function(
             $http, $scope, $uibModal, AlertsService, ComputeGraphService,
             DateTimeFormatService, ExplorationDataService,
             ExplorationFeaturesService, ExplorationStatesService,
             ReadOnlyExplorationBackendApiService, RouterService,
-            StateImprovementSuggestionService,
-            StateInteractionStatsService, StatesObjectFactory,
-            UrlInterpolationService, IMPROVE_TYPE_INCOMPLETE) {
-          var ctrl = this;
-          var _EXPLORATION_STATS_VERSION_ALL = 'all';
-          var stateStatsModalIsOpen = false;
+            StateImprovementSuggestionService, StateInteractionStatsService,
+            StatesObjectFactory, UrlInterpolationService,
+            IMPROVE_TYPE_INCOMPLETE) {
+          const ctrl = this;
+          const newCompletionRatePieChartOptions = title => ({
+            chartAreaWidth: 240,
+            colors: ['#d8d8d8', '#008808', 'blue'],
+            height: 270,
+            left: 20,
+            legendPosition: 'right',
+            pieHole: 0.6,
+            pieSliceBorderColor: 'black',
+            pieSliceTextStyleColor: 'black',
+            title: title,
+            width: 240,
+          });
 
-          ctrl.getLocaleAbbreviatedDatetimeString = function(millisSinceEpoch) {
-            return DateTimeFormatService.getLocaleAbbreviatedDatetimeString(
-              millisSinceEpoch);
+          const loadLatestExploration = () => {
+            return ReadOnlyExplorationBackendApiService.loadLatestExploration(
+              ExplorationDataService.explorationId
+            ).then(response => {
+              const { init_state_name: initStateName, states } = (
+                response.exploration);
+              ctrl.states = (
+                StatesObjectFactory.createFromBackendDict(states));
+              ctrl.statsGraphData = (
+                ComputeGraphService.compute(initStateName, ctrl.states));
+              ctrl.playthroughsAreAvailable = (
+                ExplorationFeaturesService.isPlaythroughRecordingEnabled() &&
+                !ExplorationFeaturesService.isImprovementsTabEnabled());
+            });
```

Comment:
Sorry, @brianrodri -- I can't follow the changes in this file at all. Everything seems to have moved around completely and I am having significant trouble trying to figure out what the diff is. Could you please restrict the edits to single-line or essential changes (or do whatever is needed to make the diff readable)? If you need to move things around, let's do a separate PR for that, but don't move and rewrite at the same time.

Reply:
OK, I've cut down significantly the changes (but still tried to prioritize deleting unused code).